### PR TITLE
headset suit storage sprites

### PR DESCRIPTION
**Pull request by itseasytosee**
## About The Pull Request
Makes all headsets use the default headset suit storage sprite (as opposed to no sprite)
## Why It's Good For The Game
puple checkerboard make me sad
## Changelog
:cl: itseasytosee
imageadd: missing headset sprites
/:cl:

### CODE REVIEW

File: code/game/objects/items/devices/radio/headset.dm [hunk 1]
```diff
@@ -18,6 +18,7 @@ GLOBAL_LIST_INIT(channel_tokens, list(
 	name = "radio headset"
 	desc = "An updated, modular intercom that fits over the head. Takes encryption keys."
 	icon_state = "headset"
+	worn_icon = "headset"
```
Comment (ShizCalev) [inline code/game/objects/items/devices/radio/headset.dm:21]: This isn't quite that simple for headsets, sadly.

**Review by ShizCalev — CHANGES_REQUESTED** [review]

File: code/game/objects/items/devices/radio/headset.dm (hunk 1 above)
Comment (itseasytosee) [reply code/game/objects/items/devices/radio/headset.dm:21]: > worn_icon is always an icon file. This isn't correct in the slightest.

So true

**Comment by itseasytosee** [comment]
👍

**Review by ShizCalev — CHANGES_REQUESTED** [review]
this doesn't actually do what you say it does since every subtype has it's own worn_icon_state already.

```
   	6 - /obj/item/radio/headset/syndicate/alt using invalid worn_icon_state, "syndie_headset"
  	7 - /obj/item/radio/headset/headset_sec using invalid worn_icon_state, "sec_headset"
  	8 - /obj/item/radio/headset/headset_sec/alt using invalid worn_icon_state, "sec_headset_alt"
  	9 - /obj/item/radio/headset/headset_eng using invalid worn_icon_state, "eng_headset"
  	10 - /obj/item/radio/headset/headset_rob using invalid worn_icon_state, "rob_headset"
  	11 - /obj/item/radio/headset/headset_med using invalid worn_icon_state, "med_headset"
  	12 - /obj/item/radio/headset/headset_sci using invalid worn_icon_state, "sci_headset"
  	13 - /obj/item/radio/headset/headset_medsci using invalid worn_icon_state, "medsci_headset"
  	14 - /obj/item/radio/headset/headset_srvsec using invalid worn_icon_state, "srvsec_headset"
  	15 - /obj/item/radio/headset/headset_com using invalid worn_icon_state, "com_headset"
  	16 - /obj/item/radio/headset/heads/captain/alt using invalid worn_icon_state, "com_headset_alt"
  	17 - /obj/item/radio/headset/headset_cargo using invalid worn_icon_state, "cargo_headset"
  	18 - /obj/item/radio/headset/headset_cargo/mining using invalid worn_icon_state, "mine_headset"
  	19 - /obj/item/radio/headset/headset_srv using invalid worn_icon_state, "srv_headset"
  	20 - /obj/item/radio/headset/headset_cent using invalid worn_icon_state, "cent_headset"
  	21 - /obj/item/radio/headset/headset_cent/alt using invalid worn_icon_state, "cent_headset_alt"
```

you'll need to generate actual sprites instead of relying on a one-line-change to "fix" this.

**Comment by itseasytosee** [comment]
> this doesn't actually do what you say it does since every subtype has it's own worn_icon_state already.

image
This isn't correct in the slightest. They subtypes don't have worn_icon_states declared so they default to the icon_state. Declaring the worn_icon_state in the parent does indeed fix this. Don't Pied Piper me on this.

Also, as far I am aware, the only suit you can put headsets in is the santa suit. Which, in a streak of brilliance, decided to allow all subtypes of obj/item, meaning now all items need a valid suit storage sprite. Hard to blame the santa suit maker, because at the time we didn't have the unit tests or the funky purple squares.

**Comment by optimumtact** [comment]
>Don't Pied Piper me on this.

the attitude isn't useful, they're trying to help

**Comment by itseasytosee** [comment]
👍

**Comment by ShizCalev** [comment]
image

**Comment by ShizCalev** [comment]
literally everything you said is wrong.

the unit test also filters out the santa suit for that very reason.

image

**Comment by ShizCalev** [comment]
image
image
image

headsets are a subtype of /obj/item/radio, ergo these are the jackets which allow them.

**Pull request closed without merging**